### PR TITLE
Batch create pre designed NFT collection using CLI

### PR DESCRIPTION
**Pull request by h0ngcha0**
- Add command to batch generate images using OpenAI API
- Add command to upload images to IPFS
- Add command to upload images metadata to IPFS to create pre designed collection

Workflow:
1. (Optional) Generate `n` number of images using OpenAI API based on a prompt. Not needed if you already designed the images. (`cli generate-images-with-openai`)
2. Upload the directory of `n` number of images to IPFS, a metadata json file is generated as a result (`cli upload-images-to-ipfs`)
3. Update the generated metadata file, specifically the name and description of the individual NFT
4. Upload the images metadata to IPFS, a `tokenBaseUrl` is returned (`cli upload-images-metadata-to-ipfs`)

The `tokenBaseUrl` can then be used in the NFT marketplace to create the pre-designed collection (https://github.com/alephium/alephium-nft/pull/17).

### CODE REVIEW

File: packages/cli/scripts/pre-designed-nft.ts [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+const validMetadataAttributeTypes = ['string', 'number', 'boolean']
+
+export async function generateImagesWithOpenAI(
+  openaiApiKey: string,
+  prompt: string,
+  numberOfImages: number,
+  imageSize: CreateImageRequestSizeEnum,
+  storedDir: string
+) {
+  if (!fs.existsSync(storedDir)) {
```
Comment (Lbqds) [inline packages/cli/scripts/pre-designed-nft.ts:45]: How about creating the directory if it does not exist?

File: packages/cli/scripts/pre-designed-nft.ts [hunk 2]
```diff
@@ -0,0 +1,231 @@
...
+
+    validateStringTypeIfExists(image, metadata, 'name')
+    validateStringTypeIfExists(image, metadata, 'description')
+    validateAttributesTypeIfExists(image, metadata['attributes'])
+  })
+
+  return config as NFTMetadataConfig
+}
+
+export async function validateMetadataFile(tokenBaseUri: string, maxSupply: number): Promise<NFTMetadata[]> {
```
Comment (Lbqds) [inline packages/cli/scripts/pre-designed-nft.ts:141]: This function is the same as `validateTokenBaseUri`.

File: packages/cli/scripts/pre-designed-nft.ts (hunk 1 above)
Comment (h0ngcha0) [reply packages/cli/scripts/pre-designed-nft.ts:48]: Fixed, thanks :+1:

File: packages/cli/scripts/pre-designed-nft.ts (hunk 2 above)
Comment (h0ngcha0) [reply packages/cli/scripts/pre-designed-nft.ts]: This is not needed any more, removed :+1:

File: packages/cli/scripts/pre-designed-nft.ts [hunk 3]
```diff
@@ -0,0 +1,244 @@
...
+  if (attributes) {
+    return attributes.map((attribute) => {
+      const traitType = Object.keys(attribute)[0]
+      const value = attribute[traitType]
+      return { trait_type: traitType, value }
+    })
+  }
+}
+
+function padding(num: number, size: number): string {
```
Comment (Lbqds) [inline packages/cli/scripts/pre-designed-nft.ts:234]: Nit: We can use the `padStart` function: `return num.toString().padStart(size, '0')`

**Review by Lbqds — APPROVED** [review]
LGTM 👍

File: packages/cli/scripts/pre-designed-nft.ts (hunk 3 above)
Comment (h0ngcha0) [reply packages/cli/scripts/pre-designed-nft.ts:234]: Sounds good :+1:

**Pull request merged**